app/controllers/oidc/concerns/trusted_publisher_controller.rb

```diff
@@ -0,0 +1,61 @@
+module OIDC::Concerns::TrustedPublisherController
+  extend ActiveSupport::Concern
+
+  included do
+    include SessionVerifiable
+    verify_session_before
+
+    before_action :create_params, only: %i[create]
+    before_action :set_trusted_publisher_type, only: %i[create]
+    before_action :set_page, only: :index
+  end
+
+  def create_params
+    @create_params ||=
+      params
+        .permit(
+          create_params_key => [
+            :trusted_publisher_type,
+            {
+              trusted_publisher_attributes: %i[
+                repository_owner repository_name workflow_filename environment
+              ]
+            },
+            *extra_create_params
+          ]
+        )
+        .require(create_params_key)
+  end
+
+  def extra_create_params
+    []
+  end
+
+  def create_params_key
+    raise NotImplementedError
+  end
+
+  def find_github_repository_owner_id
+    attributes = create_params.require(:trusted_publisher_attributes)
+    attributes.delete(:environment) if attributes[:environment].blank?
+    repository_owner = attributes[:repository_owner]
+    return if repository_owner.blank?
+    attributes[:repository_owner_id] =
+      begin
+        Octokit::Client.new.user(repository_owner).id
+      rescue Octokit::NotFound
+        nil
+      end
```

Comment:
I'm not a fan of this business here. `create_params` usually just "permit" a few attributes and doesn't modify them, especially with an API call.

I'd rather see this complexity shifted into the model. Maybe a class method on GithubActions that returns some additional attributes when given the inputs here? This seems like the scope of the github actions model.

Reply:
I'm really not a fan of putting any methods that touch the network onto models, I feel like they do belong in the controller layer